File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/java/io/github/lishangbu/avalon/oauth2/common/log/AuthenticationLogRecord.java
package io.github.lishangbu.avalon.oauth2.common.log;

import java.time.Instant;
import org.jspecify.annotations.Nullable;

public record AuthenticationLogRecord(
        @Nullable String username,
        @Nullable String clientId,
        @Nullable String grantType,
        @Nullable String ip,
        @Nullable String userAgent,
        boolean success,
        @Nullable String errorMessage,
        @Nullable Instant timestamp) {}

File: avalon-support/avalon-oauth2-support/avalon-oauth2-common/src/main/kotlin/io/github/lishangbu/avalon/oauth2/common/log/AuthenticationLogRecord.kt
package io.github.lishangbu.avalon.oauth2.common.log

import java.time.Instant

@JvmRecord
data class AuthenticationLogRecord(
    val username: String?,
    val clientId: String?,
    val grantType: String?,
    val ip: String?,
    val userAgent: String?,
    val success: Boolean,
    val errorMessage: String?,
    val timestamp: Instant?,
)
